sdk/keyvault/azure-security-keyvault-keys/src/key_vault_key.cpp

```diff
@@ -27,22 +28,36 @@ void ParseStringOperationsToKeyOperations(
 }
 } // namespace
 
-KeyVaultKey _detail::KeyVaultKeyDeserialize(
+KeyVaultKey _detail::KeyVaultKeySerializer::KeyVaultKeyDeserialize(
     std::string const& name,
     Azure::Core::Http::RawResponse const& rawResponse)
 {
   KeyVaultKey key(name);
-  _detail::KeyVaultKeyDeserialize(key, rawResponse);
+  _detail::KeyVaultKeySerializer::KeyVaultKeyDeserialize(key, rawResponse);
   return key;
 }
 
-void _detail::KeyVaultKeyDeserialize(
+KeyVaultKey _detail::KeyVaultKeySerializer::KeyVaultKeyDeserialize(
+    Azure::Core::Http::RawResponse const& rawResponse)
+{
+  KeyVaultKey key;
+  _detail::KeyVaultKeySerializer::KeyVaultKeyDeserialize(key, rawResponse);
+  return key;
+}
+
+void _detail::KeyVaultKeySerializer::KeyVaultKeyDeserialize(
     KeyVaultKey& key,
     Azure::Core::Http::RawResponse const& rawResponse)
 {
-  auto body = rawResponse.GetBody();
+  auto& body = rawResponse.GetBody();
```

Comment:
We shouldn't be modifying the body.  Be explicit and retrieve it const&.

```suggestion
  auto const& body = rawResponse.GetBody();
```